### ISSUE TITLE
torch.nn.LocalResponseNorm returns +/-Inf when k=0 and alpha=0 (zero denominator, no validation)

### ISSUE THREAD

**Issue by gdrosa** [pytorch/pytorch#187177]
### 🐛 Describe the bug

`torch.nn.LocalResponseNorm(size, alpha, beta, k)` computes `out = x * (k + (alpha/size) * sum(x^2))^(-beta)`. When a user sets both `k = 0` and `alpha = 0` the bracketed denominator is exactly `0`, and `0^(-beta) = +inf` for any `beta > 0`, so every output element becomes `+/-inf`. No warning or exception is raised.

This is a joint-zero corner: `k = 0` alone (with `alpha > 0`) is safe, and `alpha = 0` alone (with `k > 0`) is safe; only zeroing both at once collapses the denominator. The defaults (`alpha = 1e-4, beta = 0.75, k = 1.0`) are safe, but the constructor accepts any float for `k`, `alpha`, `beta`, `size` with no validation, so it is easy to hit while sweeping normalization hyperparameters.

```python
import torch

torch.manual_seed(0)
x = torch.randn(2, 8, 4, 4, dtype=torch.float32)
lrn = torch.nn.LocalResponseNorm(size=1, alpha=0.0, beta=0.2, k=0.0)
out = lrn(x)
print("any inf:", torch.isinf(out).any().item())
print(out.flatten()[:8])
```

Observed:

```
any inf: True
tensor([-inf, -inf, -inf, -inf,  inf,  inf, -inf, -inf])
```

Expected:

A constructor-time `ValueError` when `k + alpha <= 0`, since the LRN denominator must stay positive for `^(-beta)` to be finite. Right now the degenerate case is silently turned into `+/-inf`.

### Versions

```
PyTorch version: 2.5.1+cu121
Is debug build: False
CUDA used to build PyTorch: 12.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: 18.1.3 (1ubuntu1)
CMake version: version 3.28.3
Libc version: glibc-2.39

Python version: 3.10.20 (main, Mar 11 2026, 17:46:40) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-6.8.0-124-generic-x86_64-with-glibc2.39
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1660 SUPER
Nvidia driver version: 535.309.01
cuDNN version: Could not collect
Is XNNPACK available: True

CPU:
Architecture:                            x86_64
CPU(s):                                  20
Model name:                              12th Gen Intel(R) Core(TM) i7-12700
Thread(s) per core:                      2
Core(s) per socket:                      12

Versions of relevant libraries:
[pip3] numpy==2.2.6
[pip3] torch==2.5.1+cu121
[pip3] triton==3.1.0
```

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @malfet

**Comment by rockygeekz** [comment]
Raised a fix in #187242. The approach validates that k and alpha are not both non-positive at construction time (and in the functional API), since that combination makes the denominator zero and silently produces inf.

**Comment by rockygeekz** [comment]
Raised #187242 for this. The fix validates that k and alpha aren't both non-positive when beta is non-zero, since that combination collapses the denominator to zero. Reproduced and verified locally on torch 2.12.0+cpu.

Let me know if this needs the actionable label first or if the approach needs any changes.
